### ISSUE TITLE
Provide PTRACE_PEEKUSER API

### ISSUE THREAD

**Issue by alex** [nix-rust/nix#980]
Right now in order to use `PTRACE_PEEKUSER` you need to directly invoke `ptrace::ptrace`, which is deprecated. It'd be great if there was a non-deprecated wrapper for `PEEKUSER`.

**Comment by Susurrus** [comment]
Indeed, the goal is to get rid of the `ptrace()` function completely in favor of wrappers, and this would be a most welcome addition. Feature requests are mostly done by volunteer contributors like yourself. Would you be willing to submit a PR providing this implementation?
